cli/worker.rs

```diff
@@ -344,12 +347,39 @@ impl CliMainWorkerFactory {
     let (main_module, is_main_cjs) = if let Ok(package_ref) =
       NpmPackageReqReference::from_specifier(&main_module)
     {
+      // For main_module that exists in package.json deps,
+      // use the version defined in package.json instead of adding new dependency.
+      let existing_package_req = shared
```

Comment:
Instead of handling an "existing" vs "provided" package req downstream, perhaps this should just be the resolved `NpmPackageReqReference` and then the conditional logic below can be removed.